### DISCUSSION TITLE
is pocketbase good in production?

### DISCUSSION THREAD

**Discussion by mckoda09** [pocketbase/pocketbase#4032] [Q&A] [answered]
should i use pocketbase in production?

**Comment by ganigeorgiev** [comment, marked as answer]
> should i use pocketbase in production?

If you are not sure, then the safest short answer would be No.

Some other similar discussions you could check - https://github.com/pocketbase/pocketbase/discussions/2163, https://github.com/pocketbase/pocketbase/discussions/1970, https://github.com/pocketbase/pocketbase/discussions/1528, https://github.com/pocketbase/pocketbase/discussions/1290, https://github.com/pocketbase/pocketbase/discussions/424, etc.

It really depends on your use case and whether you are OK with applying some manual migration steps from time to time because **backwards compatibility is not guaranteed before reaching v1**.

It is a hobby project, there is no team or company behind it and, while I try to help with questions, I make no promises and cannot provide dedicated support.

PocketBase embeds SQLite and it inherits some of its limitations, see [SQLite limits](https://www.sqlite.org/limits.html). You may also find useful [Appropriate Uses For SQLite](https://www.sqlite.org/whentouse.html).

I'm planning to release Presentator v3 (my other open source project), which is based on PocketBase, in the next ~1 month and will share some numbers to help users get a general idea of what load it can handle and what to expect in terms of a performance, but until then you can explore the benchmarks repo - https://github.com/pocketbase/benchmarks.

Overall, if your use case is write heavy, needs high availability, horizontal scaling (aka. a single server is not enough), then PocketBase may not be suitable for it. In this case, you can explore some of the other similar but more mature solutions like:
- [Supabase](https://supabase.com/)
- [Appwrite](https://appwrite.io/)
- [Nhost](https://nhost.io/)
- [EdgeDB](https://www.edgedb.com/)
- [SurrealDB](https://surrealdb.com/)
- [PayloadCMS](https://payloadcms.com/)
- [Strapi](https://strapi.io/)
- [Directus](https://directus.io/)
- and probably others...

**Reply by jimafisk** [reply]
We use PocketBase for a client app with over 15k users, it's very practical to self-host and the JS SDK is a single package that's easy to understand + the documentation is great. We started the project with Supabase but switched early on to PocketBase to simplify our tech stack and we're really happy with the outcome.

**Reply by jonshipman** [reply]
That's a nice number. What are the machine specs you're using? I've been spinning up instances for clients, replacing their spreadsheet solutions with a database solution, but in my case there are probably fewer than 5 concurrent users for each.

**Reply by jimafisk** [reply]
We host PocketBase separate from the frontend, but it runs on a 2 GB Memory / 2 Intel vCPUs Digital Ocean droplet. I think it's overkill honestly, but we'd rather be safe than sorry.

**Reply by henrikzabel** [reply]
@jimafisk do you also make use of the realtime connections?

**Reply by jimafisk** [reply]
@HenrikZabel - We're not using realtime, so I don't have insights on that unfortunately.
